src/core/SocketAdapter.js

```
@@ -54,6 +64,46 @@ export default function SocketAdapter(client) {
     }
   }
 
+  function handleFragment(fragment) {
+    const { id, data, num, total } = fragment;
+    if (!id || typeof num !== 'number' || typeof total !== 'number' || typeof data !== 'string') {
+      // Invalid fragment, ignore
+      return;
+    }
+    // If total is a float, use its integer part for fragment count
+    const totalInt = Math.floor(total);
+    if (!fragmentBuffer.has(id)) {
+      fragmentBuffer.set(id, { fragments: [], received: 0, total: totalInt });
+    }
+    const entry = fragmentBuffer.get(id);
+
+    if (!entry) {
+      // Should not happen, keeping TypeScript happy
+      return;
```

The one minor non-blocking comment I have here is that this should probably throw an exception. I think we all know that things that should never happen tend to happen 😉